### PR TITLE
Modify VMware tests to run on VMware8

### PR DESCRIPTION
**Pull request by shubhamsg199**
Adding support to run VMware tests on VMware8

### CODE REVIEW

File: pytest_fixtures/component/provision_vmware.py [hunk 1]
```diff
@@ -4,12 +4,23 @@
 from robottelo.config import settings
 
 
+@pytest.fixture
+def vmware_version(request):
+    versions = {
+        'vmware7': settings.vmware.vcenter7.hostname,
+        'vmware8': settings.vmware.vcenter8.hostname,
+    }
+    return versions[getattr(request, 'param', 'vmware7')]
```
Comment (Gauravtalreja1) [inline pytest_fixtures/component/provision_vmware.py:13]: Even though it's not important here, keep always the latest :)
```suggestion
    return versions[getattr(request, 'param', 'vmware8')]
```

File: pytest_fixtures/component/provision_vmware.py [hunk 2]
```diff
@@ -4,12 +4,23 @@
 from robottelo.config import settings
 
 
+@pytest.fixture
+def vmware_version(request):
```
[the remark this answers is not in the document]
Comment (shubhamsg199) [reply pytest_fixtures/component/provision_vmware.py:8]: How about `vmware_hostname`  ?

File: tests/foreman/ui/test_computeresource_vmware.py [hunk 3]
```diff
@@ -562,7 +571,7 @@ def test_positive_virt_card(session, target_sat, module_location, module_org):
         assert virt_card['cluster'] == settings.vmware.cluster
         assert virt_card['memory'] == '5 GB'
         assert 'public_ip_address' in virt_card
-        assert virt_card['mac_address'] == settings.vmware.mac_address
+        assert virt_card['mac_address'] != ''
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_computeresource_vmware.py:574]: I think we should assert here with actual mac address check here and we can run this test on either version as there'll be no change in virt card displayed, right?

Comment (shubhamsg199) [reply tests/foreman/ui/test_computeresource_vmware.py]: Updated.

File: conf/vmware.yaml.template [hunk 4]
```diff
@@ -16,8 +23,6 @@ VMWARE:
   VM_NAME:
   # MAC_ADDRESS: Mac address of the vm
   MAC_ADDRESS:
```
[the remark this answers is not in the document]
Comment (shubhamsg199) [reply conf/vmware.yaml.template]: Yeah, updated

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_computeresource_vmware.py::test_positive_vmware_cr_end_to_end tests/foreman/ui/test_computeresource_vmware.py

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_computeresource_vmware.py::test_positive_vmware_cr_end_to_end tests/foreman/ui/test_computeresource_vmware.py

File: conf/vmware.yaml.template [hunk 5]
```diff
@@ -1,7 +1,14 @@
 VMWARE:
   # VMware to be added as a compute resource
   # VCENTER: vmware vcenter URL
-  VCENTER:
+  VCENTER7:
+    HOSTNAME:
+    HYPERVISOR:
+    MAC_ADDRESS:
```
Comment (Gauravtalreja1) [inline conf/vmware.yaml.template:7]: Could you add a comment here for mac addr field to know which vm it belongs or vm_name filed in this template?

Comment (shubhamsg199) [reply conf/vmware.yaml.template:8]: Updated

File: conf/vmware.yaml.template [hunk 6]
```diff
@@ -1,7 +1,16 @@
 VMWARE:
   # VMware to be added as a compute resource
   # VCENTER: vmware vcenter URL
-  VCENTER:
+  VCENTER7:
+    HOSTNAME:
+    HYPERVISOR:
+    # mac_address: Mac address of the automation-robottelo-vmware-vm
```
Comment (Gauravtalreja1) [inline conf/vmware.yaml.template:7]: vm_name we're not defining here in template
```suggestion
    # mac_address: Mac address of the vm_name
```

**Review by Gauravtalreja1 — APPROVED** [review]
@shubhamsg199 Thanks for addressing the comments, great job on this!

**Pull request merged**